app/src/utils/apiHelpers.test.tsx

```diff
@@ -0,0 +1,212 @@
+import {
+  getDivisionData,
+  getFeaturedImage,
+  getFeaturedItemData,
+  getHomePageData,
+  getItemsCountFromUUIDs,
+  getNumDigitizedItems,
+  getRandomFeaturedItem,
+} from "./apiHelpers";
+import { apiResponse } from "./apiResponse";
+import defaultFeaturedItem from "../data/defaultFeaturedItemData";
+import {
+  mockFeaturedItemResponse,
+  mockItemResponse,
+} from "__tests__/__mocks__/data/mockApiResponses";
+
+jest.mock("./apiResponse");
+
+beforeEach(() => {
+  jest.clearAllMocks();
+});
+
+describe("getHomePageData", () => {
+  it("creates response containing random number and all 7 lanes", async () => {
+    (apiResponse as jest.Mock).mockResolvedValueOnce(
+      Promise.resolve({
+        nyplAPI: {
+          response: {
+            counts: {
+              count: [
+                {
+                  uuid: { $: "test" },
+                  count_value: { $: "10" },
+                },
+              ],
+            },
+          },
+        },
+      })
+    );
+    const result = await getHomePageData();
+    expect(apiResponse as jest.Mock).toHaveBeenCalled();
+    expect([0, 1]).toContain(result.randomNumber);
+    expect(result.lanesWithNumItems.length).toEqual(7);
+    // Test uuid response doesn't match the collection uuid, so all of these should be 0.
+    expect(
+      result.lanesWithNumItems[0].collections[3].numberOfDigitizedItems
+    ).toEqual("0");
+  });
+});
+
+describe("getFeaturedItemData", () => {
+  it("creates response containing featuredItem and numDigitizedItems", async () => {
+    const result = await getFeaturedItemData();
+
+    // Fallback values:
+    expect(result.numberOfDigitizedItems).toEqual("875,861");
+    expect(result.featuredItem.imageID).toEqual(
+      defaultFeaturedItem.production.featuredItem.imageID
+    );
+  });
+});
+
+describe("getDivisionData", () => {
+  it("forms the correct request from params", async () => {
+    const result = await getDivisionData({
+      slug: "testSlug",
+      pageNum: 1,
+      perPage: 3,
+    });
+
+    expect(apiResponse as jest.Mock).toHaveBeenCalledWith(
+      `${process.env.API_URL}/api/v2/divisions/testSlug?page=1&per_page=3`
+    );
+  });
+  it("returns successful response", async () => {
```

Comment:
What about when the `apiResponse` throws an error? How does `getDivisionData` handle it? Add a test.

Reply:
I'm not sure how to go about doing this since `getDivisionData` doesn't catch or handle at all– the error handling is all in `apiResponse`.  Is that a flaw in these getters ?

Reply:
Yes, it's a bit of a flaw. We decided to let Next.js catch the thrown error that bubbles up in its own error page, so the app itself is handling this appropriately (e.g. the app does not crash or throw the unstyled 500 error page). If we catch the error and don't do anything, the app won't display the error page.

Coding-wise, these functions cannot really be used outside of this app, not that we would though. But if you're using a function like this outside of this nextjs app and it throws an error and you don't catch it, it is perhaps unexpected. You can (and should) type the functions or add more jsdoc/esdoc style comments to say explicitly it returns a data object or throws an error, `const getDivisionData : [typeof apiResponse] | Error = async ({ ...`

Again, we made this tradeoff and went with Next handling the thrown errors so overall, we're ok. For function completeness though, you can add a test for the thrown error that is similar to the `apiResponse` timeout  error
```
await expect(apiCall).rejects.toEqual(
      new Error("apiResponse: Request timed out")
    );
```

This is a very interesting topic so we can keep talking about this. This is an ecmascript proposal but I really like this idea: https://github.com/arthurfiorette/proposal-safe-assignment-operator
You'd end up with `const [error, response] ?= await fetch("https://api.example.com/data")` and similar for these getters.

Instead of throwing an error, the function returns the error and the response. Then at a higher level you can throw the error to let Nextjs handed it. It's lots of layers but this would be a good repo-wide project to work on. 

To be clear, for now, an error test is enough.